packages/shared/src/react/types.ts

```diff
@@ -12,11 +19,15 @@ export type PaginatedResources<T = unknown> = {
   fetchNext: () => void;
   hasNextPage: boolean;
   hasPreviousPage: boolean;
+  revalidate: () => Promise<void>;
+  setData: Infinite extends true
+    ? // Array of pages of data
+      CacheSetter<(ClerkPaginatedResponse<T> | undefined)[]>
```

Comment:
❓ shouldn't this be ` CacheSetter<(ClerkPaginatedResponse<T>[] | undefined)>` ?

Reply:
@dimkl a "page" in swr can be either undefined or has a value of type ClerkPaginatedResponse<T>